### PR TITLE
feat: Smoke test

### PR DESCRIPTION
**Pull request by JiaYongfei**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/platform/platform_test.go [hunk 1]
```diff
@@ -37,65 +39,248 @@ import (
...
+		var credential platformv1.ClusterCredential
+
+		BeforeEach(func() {
+			if cluster == nil {
+				// Prepare a cluster to be imported
+				cluster, err = createCluster(1)
+				Expect(err).To(BeNil())
+
+				// Get the credential of cluster
+				credentials, err := tkeClient.PlatformV1().ClusterCredentials().List(context.Background(), metav1.ListOptions{
```
Comment (huxiaoliang) [inline test/e2e/platform/platform_test.go:239]: why not use `Get` api instead of `List`？

File: test/e2e/platform/platform_test.go [hunk 2]
```diff
@@ -37,65 +39,248 @@ import (
...
+
+			It("Add node to cluster", func() {
+				for _, machine := range machines {
+					out, err := runCmd("kubectl get mc --kubeconfig " + tkeKubeConfigFile + " | grep " + machine.Name)
+					Expect(err).Should(BeNil())
+					Expect(out).Should(ContainSubstring("Running"))
+				}
+			})
+
+			It("Add tag to node", func() {
```
Comment (huxiaoliang) [inline test/e2e/platform/platform_test.go:120]: should be `add label to node`

Comment (JiaYongfei) [reply test/e2e/platform/platform_test.go:120]: Let me update it in next commit. Thanks

File: test/e2e/platform/platform_test.go (hunk 1 above)
Comment (JiaYongfei) [reply test/e2e/platform/platform_test.go:239]: Let me check if we can get the credential name or not from cluster. If yes, I'll update it to use Get instead

File: test/e2e/platform/platform_test.go [hunk 3]
```diff
@@ -37,65 +39,248 @@ import (
...
-		}
-		for i, one := range workerNodes {
-			fmt.Printf("delete instance %d %s\n", i, one.InternalIP)
-			instanceIDs = append(instanceIDs, &workerNodes[i].InstanceID)
-		}
-		err := provider.DeleteInstances(instanceIDs)
-		Expect(err).To(BeNil())
+		It("Create Baremetal cluster", func() {
+			out, err := runCmd("kubectl get clusters --kubeconfig " + tkeKubeConfigFile + " | grep " + clsName)
+			Expect(err).Should(BeNil())
```
Comment (huxiaoliang) [inline test/e2e/platform/platform_test.go:89]: if you depends on `kubectl`  to create cluster, do you mean we need install kubectl on the CI host? how to handle kubectl upgrade?  why node use go api to do this?

**Review by huxiaoliang — CHANGES_REQUESTED** [review]

File: test/e2e/platform/platform_test.go (hunk 3 above)
Comment (JiaYongfei) [reply test/e2e/platform/platform_test.go:89]: It doesn't depend on kubectl to create cluster. kubectl was introduced to do some check as mentioned in the manual test cases.

Comment (huxiaoliang) [reply test/e2e/platform/platform_test.go:89]: why need `kubectl`, who install it on CI host? why not use go-clinet/go-api to do check?

File: test/e2e/platform/platform_test.go [hunk 4]
```diff
@@ -37,65 +39,248 @@ import (
...
+
+func createClusterWithMasterNodes(masterNodes []cloudprovider.Instance) (cluster *platformv1.Cluster, err error) {
+	klog.Info("Create cluster")
+	cluster = &platformv1.Cluster{
 		Spec: platformv1.ClusterSpec{
-			Type:     "Baremetal",
-			Features: platformv1.ClusterFeature{EnableMasterSchedule: true},
+			Type:          "Baremetal",
+			Features:      platformv1.ClusterFeature{EnableMasterSchedule: true},
+			Version:       "1.18.3",
```
Comment (leoryu) [inline test/e2e/platform/platform_test.go:281]: Hard code cluster version is not a good way. If next release doesn't support this version, test will fail.

Comment (huxiaoliang) [reply test/e2e/platform/platform_test.go:281]: nice catch!  cool!

Comment (huxiaoliang) [reply test/e2e/platform/platform_test.go:281]: pkg/spec/spec.go has K8sVersions to fix it, please use the latest version for e2e test

File: test/e2e/platform/platform_test.go (hunk 3 above)
Comment (JiaYongfei) [reply test/e2e/platform/platform_test.go:91]: As discussed offline, I'll remove `kubectl` related code, but make sure a kubeconfig file generated there for easily use if needed.

File: test/e2e/platform/platform_test.go (hunk 4 above)
Comment (JiaYongfei) [reply test/e2e/platform/platform_test.go]: Got it. Thx

**Pull request closed without merging**